Update to align with Cookiecutter-Nautobot-App v2.4.2

smk4664 opened the pull request:
As part of adding this app to Design Builder, we are updating it to the latest 2.4.2 release of the Cookiecutter template.

Note: I set the Check-in docker to minimum 2.2.0 of Nautobot due to a Migration relying on an a Core migration that was not implemented until Nautobot 2.2.0

smk4664 committed:
Update to align with Cookiecutter-Nautobot-App v2.4.2

smk4664 commented on tasks.py:
```diff
@@ -777,7 +838,6 @@ def unittest(
         command += " --verbosity 2"
 
     run_command(context, command)
-    run_command(context, "coverage lcov --include 'nautobot_design_builder/*' -o lcov.info")
```
This will be coming back in cookiecutter-nautobot-app 2.5.0, but not part of the unitest, it will have it's own step.

smk4664 committed:
Correct url to docs

smk4664 committed:
Fix Ruff

I am sorry

smk4664 committed:
Update poetry.lock

smk4664 committed:
Set minimum to match the bgp-models minimum.

smk4664 committed:
Minimum is 2.2.0 due to migrations, picking a slightly higher version.

chadell reviewed (approved):
LGTM

chadell commented on .github/workflows/ci.yml:
```diff
@@ -92,7 +92,7 @@ jobs:
       matrix:
         python-version: ["3.11"]
-        nautobot-version: ["2.0.3"]
+        nautobot-version: ["2.2.3"]
```
should we update the plugin requirements to be specific that this is the minimum Nautobot version?

smk4664 committed:
Set minimum version to match the migration

In Design-Builder 2.1.0, a migration was implemented that required a migration from Nautobot that was not introduced until Nautobot 2.2.0
